### ISSUE TITLE
[Feature request] Table options/attributes for events

### ISSUE THREAD

**Issue by Nevets82** [wenzhixin/bootstrap-table#4706]
I would like to have table options/attributes for events, like: data-on-load-success or data-on-load-error. The value of the options/attributes would be the name of the function like, myOnLoadSuccess or myOnLoadError.

This feature is already almost implemented but is missing a fundamental test/cast.

Here is an example of what I am trying to accomplish: [https://live.bootstrap-table.com/code/Nevets82/1190](https://live.bootstrap-table.com/code/Nevets82/1190)

Currently this throws an exception on line 5531 because the bootstrap-table event is a string (which holds the name of the function to call) and not a function.

>   Uncaught TypeError: (intermediate value)(intermediate value)[BootstrapTable.EVENTS[name]].apply is not a function
    at BootstrapTable.trigger <b>(bootstrap-table.js:5531)</b>
    at Object.success (bootstrap-table.js:5408)
    at u (VM20 jquery-3.3.1.min.js:2)
    at Object.fireWith [as resolveWith] (VM20 jquery-3.3.1.min.js:2)
    at k (VM20 jquery-3.3.1.min.js:2)
    at XMLHttpRequest.<anonymous> (VM20 jquery-3.3.1.min.js:2)

I was able to work around this by adding the following lines of code before line 5531.

``` javascript
var eventFunction = this.options[BootstrapTable.EVENTS[name]];
	      
if (typeof eventFunction === "string" && typeof window[eventFunction] === "function") {
  this.options[BootstrapTable.EVENTS[name]] = window[eventFunction];
}
```

I guess the same could be done [here](https://github.com/wenzhixin/bootstrap-table/blob/36473f6612f24f541e0ff701a346bc629c1f4e76/src/bootstrap-table.js#L1710) but maybe some kind of init function would be better suited.

Please, let me know what you think about this.

**Comment by UtechtDustin** [comment]
Is there a reason why you cant use the normal event binding method ?

**Comment by Nevets82** [comment]
The problem is that I have multiple tables on a page and that I want to handle the onLoadError for each of them. If I would use normal event binding I would have to know the id of each table and create a jQuery selector to bind to the event. So this is why I would have used the data attributes for events.

But, after carefully reviewing my problem I was able to solve this by selecting all bootstrap-tables.

``` javascript
$('table[data-toggle="table"]').each(function (index) {
    $(this).bootstrapTable({
        onLoadError: function (status, jqXHR) {
          // Do something useful here...
        }
    })
});
```

So basically I would not need this feature anymore, although it could be useful and is almost implemented.

PS: Great work on this project ;-)

**Comment by UtechtDustin** [comment]
Thank you for the Feedback :-)
